### PR TITLE
feat: Add code to simplify exporting dockerhub data for edgexfoundry.

### PR DESCRIPTION
**Pull request by bill-mahoney**
Archives a csv file in Jenkins upon demand.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/edgexfoundry/ci-management/blob/master/.github/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Added labels
## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Issue Number:
https://github.com/edgexfoundry/cd-management/issues/50

## Sandbox Testing
Test Links :
https://jenkins.edgexfoundry.org/sandbox/view/All/job/Functional-Testing/job/test_docker_audit/36/console

## Are there any new imports or modules? If so, what are they used for and why?
No

## Other information

> [INFO]  Executed 11 unit tests
> [INFO]  All unit tests passed.
> [INFO]  Overall coverage is 80%
> [INFO]  Overall coverage branch coverage is 48%
> [INFO]  Overall coverage partial branch coverage is 85%
> [INFO]  Average complexity: A (2.3333333333333335)

Signed-off-by: Bill Mahoney <bill.mahoney@intel.com>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
+FROM python:3.6.5-slim
```
Comment (soda480) [inline Dockerfile:1]: consider updating to be multi-stage Alpine-based docker image to reduce image size

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,23 @@
+properties([[
```
Comment (soda480) [inline Jenkinsfile:1]: consider refactoring to declarative pipeline for consistency sake

File: dockerhub-tags-07.22.2020-23.29.07.csv [hunk 3]
```diff
@@ -0,0 +1,415 @@
+repo_name,tag_name,architecture,os,size_in_MB,repo_star_count,repo_pull_count,last_updater_username,last_updated,days_since_update
```
Comment (soda480) [inline dockerhub-tags-07.22.2020-23.29.07.csv:1]: consider moving csv into a reports directory as to not pollute root folder

File: src/main/python/dha/cli.py [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+    parser = get_parser()
+    try:
+        args = parser.parse_args()
+        validate(args)
+
+        configure_logging(args)
+
+        dockerhub_client = get_dockerhub_client(args.dockerhub_host_api)
+
+        imageDictList = get_all_images(args, dockerhub_client)
```
Comment (soda480) [inline src/main/python/dha/cli.py:314]: code leverages competing variable name styles; `lower_case` vs `mixedCase`
ideal to choose one case throughout.
https://www.python.org/dev/peps/pep-0008/#function-and-variable-names

File: Dockerfile (hunk 1 above)
Comment (jamesrgregg) [reply Dockerfile:1]: +1

Comment (jamesrgregg) [reply Dockerfile:1]: +1

File: src/main/python/dha/cli.py (hunk 4 above)
Comment (jamesrgregg) [reply src/main/python/dha/cli.py:314]: This goes back to a standard question we have yet to address as formal guidance.

File: Jenkinsfile (hunk 2 above)
Comment (ernestojeda) [reply Jenkinsfile:1]: Yes. Declarative only for edgex

File: src/unittest/python/test_cli.py [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+        logging_patch.getLogger.return_value = root_logger_mock
+        configure_logging(args)
+        root_logger_mock.addHandler.assert_called_once()
+
+    @patch('dha.cli.logging')
+    def test__configure_logging_Should_CallExpected_When_Called_With_Args(
+        self,
+        logging_patch,
+            *patches):
+        args = Mock()
```
Comment (soda480) [inline src/unittest/python/test_cli.py:66]: indentation is not distinguishable:
https://www.python.org/dev/peps/pep-0008/#indentation

**Comment by soda480** [comment]
It appears that this branch is missing our standard PR template; missing Assignee, Labels, Projects, Linked Issues, etc.

**Comment by soda480** [comment]
Can you also add the following meta-data from Python pybuilder output:

- # of Unit Tests
- Unit Test Coverage
- Cyclomatic Complexity

File: reports/dockerhub-tags-07.22.2020-23.29.07.csv [hunk 6]
```diff
@@ -0,0 +1,415 @@
+repo_name,tag_name,architecture,os,size_in_MB,repo_star_count,repo_pull_count,last_updater_username,last_updated,days_since_update
```
Comment (ernestojeda) [inline reports/dockerhub-tags-07.22.2020-23.29.07.csv:1]: I am not sure this file is needed here.

**Pull request merged**